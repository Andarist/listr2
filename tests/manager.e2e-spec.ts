import { Manager } from '@root/index'

describe('skip a task', () => {
  let manager: Manager<any, 'verbose'>
  let log: jest.SpyInstance<void, string[][]>

  beforeEach(async () => {
    manager = new Manager<any, 'verbose'>({ renderer: 'verbose' })

    log = jest.spyOn(console, 'log').mockImplementation()
  })

  afterEach(async () => {
    jest.clearAllMocks()
  })

  it('should add tasks to the manager', async () => {
    manager.add([
      {
        task: (): Promise<void> => Promise.resolve()
      }
    ])

    await manager.runAll()

    expect(log.mock.calls).toMatchInlineSnapshot(`
      [
        [
          "[STARTED] Task without title.",
        ],
        [
          "[STARTED] Task without title.",
        ],
        [
          "[SUCCESS] Task without title.",
        ],
        [
          "[SUCCESS] Task without title.",
        ],
      ]
    `)
  })

  it('should indent the task in manager', async () => {
    manager.add([
      {
        ...manager.indent([
          {
            task: (): Promise<void> => Promise.resolve()
          }
        ])
      }
    ])

    await manager.runAll()

    expect(log.mock.calls).toMatchInlineSnapshot(`
      [
        [
          "[STARTED] Task without title.",
        ],
        [
          "[STARTED] Task without title.",
        ],
        [
          "[STARTED] Task without title.",
        ],
        [
          "[SUCCESS] Task without title.",
        ],
        [
          "[SUCCESS] Task without title.",
        ],
        [
          "[SUCCESS] Task without title.",
        ],
      ]
    `)
  })

  it('should change the context in manager', async () => {
    manager.ctx = {
      test: true
    }

    manager.add([
      {
        title: 'child',
        enabled: (ctx): boolean => ctx.test,
        task: (): Promise<void> => Promise.resolve()
      }
    ])

    await manager.runAll()

    expect(log.mock.calls).toMatchInlineSnapshot(`
      [
        [
          "[STARTED] Task without title.",
        ],
        [
          "[STARTED] child",
        ],
        [
          "[SUCCESS] child",
        ],
        [
          "[SUCCESS] Task without title.",
        ],
      ]
    `)
  })
})
