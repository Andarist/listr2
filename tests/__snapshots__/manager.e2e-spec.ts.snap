// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`skip a task should add tasks to the manager: GFB8X86ruFomKEgjsONBJpAyKOAF3NXP-exit 1`] = `[]`;

exports[`skip a task should add tasks to the manager: GFB8X86ruFomKEgjsONBJpAyKOAF3NXP-stderr 1`] = `[]`;

exports[`skip a task should add tasks to the manager: GFB8X86ruFomKEgjsONBJpAyKOAF3NXP-stdout 1`] = `
[
  "{"event":"STATE","data":"STARTED","task":{"hasFinalized":false}}
",
  "{"event":"STATE","data":"STARTED","task":{"hasFinalized":false}}
",
  "{"event":"STATE","data":"COMPLETED","task":{"hasFinalized":true}}
",
  "{"event":"STATE","data":"COMPLETED","task":{"hasFinalized":true}}
",
]
`;

exports[`skip a task should change the context in manager: 5DzFgF4pfTuE12WN84l3ogCgRJPb6t9L-exit 1`] = `[]`;

exports[`skip a task should change the context in manager: 5DzFgF4pfTuE12WN84l3ogCgRJPb6t9L-stderr 1`] = `[]`;

exports[`skip a task should change the context in manager: 5DzFgF4pfTuE12WN84l3ogCgRJPb6t9L-stdout 1`] = `
[
  "{"event":"STATE","data":"STARTED","task":{"hasFinalized":false}}
",
  "{"event":"STATE","data":"STARTED","task":{"title":"child","hasFinalized":false}}
",
  "{"event":"STATE","data":"COMPLETED","task":{"title":"child","hasFinalized":true}}
",
  "{"event":"STATE","data":"COMPLETED","task":{"hasFinalized":true}}
",
]
`;

exports[`skip a task should indent the task in manager: iKVBsenKiZgkkgSQSW6OEluZqkppF7j8-exit 1`] = `[]`;

exports[`skip a task should indent the task in manager: iKVBsenKiZgkkgSQSW6OEluZqkppF7j8-stderr 1`] = `[]`;

exports[`skip a task should indent the task in manager: iKVBsenKiZgkkgSQSW6OEluZqkppF7j8-stdout 1`] = `
[
  "{"event":"STATE","data":"STARTED","task":{"hasFinalized":false}}
",
  "{"event":"STATE","data":"STARTED","task":{"hasFinalized":false}}
",
  "{"event":"STATE","data":"STARTED","task":{"hasFinalized":false}}
",
  "{"event":"STATE","data":"COMPLETED","task":{"hasFinalized":true}}
",
  "{"event":"STATE","data":"COMPLETED","task":{"hasFinalized":true}}
",
  "{"event":"STATE","data":"COMPLETED","task":{"hasFinalized":true}}
",
]
`;
